src/game.cpp

```diff
@@ -915,10 +915,10 @@ void game::create_factions()
 {
     faction tmp;
     std::vector<std::string> faction_vector = tmp.all_json_factions();
-    for(std::vector<std::string>::reverse_iterator it = faction_vector.rbegin(); it != faction_vector.rend(); ++it) {
-        tmp = faction(it[0].c_str());
+    for(auto &cur_fac : faction_vector) {
+        tmp = faction(cur_fac.c_str());
```

Comment:
And your IDE doesn't warn about this pointless use of `.c_str()`? The constructor of `faction` takes a `std::string`, not a `const char*`.

Reply:
It did actually, but I didn't flatten that particular warning at the time. I can add the few cases this shows up into the PR pretty easily here.